dashboard/src/main/java/com/google/cloud/tools/opensource/dashboard/DashboardMain.java

```diff
@@ -375,23 +390,68 @@ private static ArtifactResults generateArtifactReport(
     return upperBoundFailures;
   }
 
+  /**
+   * Returns a table where rows are jar files in {@code jarFilter}, columns are {@link
```

Comment:
Not every JAR in the filter will be in the table, right—because it's just a filter?

How about

Returns a table where the row keys are JAR files with problems, the column keys are problems, and the values are the classes in those JAR files with those problems.

If `jarFilter` is not null, the table will contain only rows for JAR files in that set.